revisions-information.md

```diff
@@ -0,0 +1,74 @@
+
+## Where can I see revisions?
+
+Packages with revisions will tell you so on their cabal page. You can always access the revisions of a package by appending `/revisions` to its url, like so: http://hackage.haskell.org/package/lens-3.5.1/revisions/
+
+## How can I download a package without revisions applied?
+
+`cabal get --pristine`
+
+## How are revisions numbered?
+
+Revisions are numbered by natural numbers, starting at zero, and increase in increments of 1.
+
+## What can revisions change?
+
+Revisions are intended to only update metadata used by build tooling, or that is used purely for display purposes. The source of a package is never changed, only specified fields of cabal files. Given the same fixed dependencies and flag configuration, as long as a metadata change hasn't disallowed them, the build product from a compiler should be the same regardless of revision changes.
+
+Specifically, revisions can alter the following:
+
+* Dependency bounds (tightening or loosening)
+* Cabal spec version (only between the interval 1.10 and 1.21)
+* Custom-setup dependencies (adding, removing or altering bounds)
+* Flag defaults
+* Flag type (automatic to manual only)
+* Flag descriptions
+* Dependencies (adding, but only in very special cases, see below).
+* Copyright attribution
+* Maintainer attribution
+* Author attribution
+* Tested-with information
+* Bug report URL
+* Homepage URL
+* Source repositories
+* Package synopsis
+* Package description
+* Package category
+
+Revisions can also add custom-setup stanzas. This is necessary because newer cabal library versions make explicit custom-setup depends, while older versions simply shared dependencies with the package proper.
+
+Other exceptions for adding dependencies all fall in the camp of needing to add things which had been implicitly assumed in the past, from a small whitelist. Package dependencies which may be added are "base" and "base-orphans". Build tool dependencies which may be added are "alex", "c2hs", "cpphs", "greencard", "happy" and "hsc2hs".
+
+
+## What can't revisions change?
+
+Revisions can't change anything else. In particular, they cannot alter code, and cannot alter anything besides the cabal file. Among some specific things revisions _cannot_ alter are:
+
+* Dependencies (removing)
+* Dependencies (adding, except as above)
+* Package name
+* Package version
+* Package license
+* Build-type
+* Modules
+* Flags (adding or removing)
+* library, sub-library, executable, test-suite, or foreign-library stanzas (adding or removing)
+
+## Why can't I revise licenses?
+
+Because once code is released under a license, it remains under that license
```

Comment:
there's an additional technical reason actually: the `LICENSE` file is included in the tarball, and typically you have to assume that the source files are also sprinkled with (c) notices; it would in theory be possible to allow to "corrent" the licence field; e.g. when we add support for SPDX, we may be able to actually make the licence declaration more accurate; however this is very difficult to automatically validate; hence we're erring on the safe side here, and avoid opening that pandora box for now.... ;-)